src/Data/Nat/Primality.agda

```diff
@@ -8,80 +8,331 @@
 
 module Data.Nat.Primality where
 
-open import Data.Empty using (⊥)
 open import Data.Nat.Base
 open import Data.Nat.Divisibility
 open import Data.Nat.GCD using (module GCD; module Bézout)
 open import Data.Nat.Properties
-open import Data.Product.Base using (∃; _×_; map₂; _,_)
-open import Data.Sum.Base using (_⊎_; inj₁; inj₂)
+open import Data.Product.Base using (_×_; map₂; _,_)
+open import Data.Sum.Base using (_⊎_; inj₁; inj₂; [_,_]′)
 open import Function.Base using (flip; _∘_; _∘′_)
 open import Relation.Nullary.Decidable as Dec
-  using (yes; no; from-yes; ¬?; decidable-stable; _×-dec_; _→-dec_)
+  using (yes; no; from-yes; from-no; ¬?; _×-dec_; _⊎-dec_; _→-dec_; decidable-stable)
 open import Relation.Nullary.Negation using (¬_; contradiction)
-open import Relation.Unary using (Decidable)
+open import Relation.Unary using (Pred; Decidable; IUniversal; Satisfiable; _∩_; _⇒_)
 open import Relation.Binary.PropositionalEquality
-  using (refl; cong)
+  using (_≡_; _≢_; refl; cong)
 
 private
   variable
-    n : ℕ
+    d k m n p : ℕ
+
+  instance
+    nt[2] : NonTrivial 2
+    nt[2] = nonTrivial {0}
```

Comment:
So it feels strange to have to explicitly declare this... why is this not caught under the generic `nonTrivial` instance?

Reply:
I'm not sure off the top of my head, but it does indicate a bug. If you can't spot the problem, let me know and I'll try and take a look.

Reply:
Commented out, it generates unsolved metas for Nats at:
* L225
* L228
* L241
* L298
* L359
* L413

all seemingly arising from constructor applications...

Reply:
So assistance with the `instance` typechecking weirdness would be *very* helpful!

Reply:
Cf. [Agda issue #6976](https://github.com/agda/agda/issues/6976)?